build/build.go

```diff
@@ -605,13 +635,12 @@ func (db *DeploymentBuilder) buildPlatformNetworks(d *Deployment) error {
 			// We are passing pointer to PlatformNetwork resource (net), hence, calling filterPlatformNetworks
 			// before or after append wouldn't make any difference and either way it would work.
 			d.PlatformNetworks = append(d.PlatformNetworks, net)
```

Comment:
It's better loop through []AddressPool, call `starlingxv1.NewAddressPool` and append the spec to `d.AddressPools` after this appending to d.PlatformNetworks.

Reply:
Changes are done